lambdas/process-journey-event/src/test/java/uk/gov/di/ipv/core/processjourneyevent/JourneyMapPageContextTest.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.di.ipv.core.processjourneyevent;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import uk.gov.di.ipv.core.library.domain.IpvJourneyTypes;
+import uk.gov.di.ipv.core.processjourneyevent.statemachine.StateMachineInitializer;
+import uk.gov.di.ipv.core.processjourneyevent.statemachine.states.BasicState;
+import uk.gov.di.ipv.core.processjourneyevent.statemachine.states.State;
+import uk.gov.di.ipv.core.processjourneyevent.statemachine.stepresponses.PageStepResponse;
+import uk.org.webcompere.systemstubs.environment.EnvironmentVariables;
+import uk.org.webcompere.systemstubs.jupiter.SystemStub;
+import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(SystemStubsExtension.class)
+public class JourneyMapPageContextTest {
+    @SystemStub private static EnvironmentVariables environmentVariables;
+
+    @BeforeAll
+    public static void beforeAll() {
+        environmentVariables.set("IS_LOCAL", "true");
+    }
+
+    private final HashMap<String, List<String>> acceptedStateContexts =
+            new HashMap<>() {
+                {
+                    put("pyi-suggest-other-options", List.of("no-photo-id"));
+                }
+            };
+
+    @ParameterizedTest
+    @EnumSource
+    void shouldHandleSameContextForSamePage(IpvJourneyTypes journeyType) throws IOException {
```

Comment:
I think that this will catch cases where someone adds an unexpected context to a page.

Is there anyway we can catch people forgetting to add a context to a page that needs one (that seems more likely to be a problem)? I suspect not, unless we force those pages always take a context e.g. add a `photo-id` possible context for `pyi-suggest-other-options` and use that in all the other places.

Reply:
> I think that this will catch cases where someone adds an unexpected context to a page.

This was what I concluded was the only useful test to have here.

The ticket originally asked if we could use ENUM values for contexts we use in the journey-map config, and this test is trying to enforce a similar behaviour (because we can't use ENUMS)

Enforcing a mandatory context isn't a bad idea, but it would involve a larger amount of refactoring everytime somebody wanted to make a page dynamic (when the purpose of the ticket is to make the process a lot smoother)

Reply:
The existing journey map tests follow a similar principle - checking that states are consistent with each other, rather than trying to define specific rules.

The rationale is that when adding new context, you aren't going to forget to add it the first one, and this test will make sure you add it to others of the same type (now and in future).

Maintaining a separate list of which pages need what context feels painful to maintain!